### PR TITLE
Switch to PPAs in Rock

### PR DESCRIPTION
**Pull request by dragomirp**
# Issue
* [DPE-1657](https://warthogs.atlassian.net/browse/DPE-1657)
* Use ppas for patroni, pgbackrest and pgbouncer

# Solution

# Context

# Testing

# Release Notes
Use ppas for patroni, pgbackrest and pgbouncer

[DPE-1657]: https://warthogs.atlassian.net/browse/DPE-1657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/integration/ha_tests/helpers.py [hunk 1]
```diff
@@ -686,11 +701,11 @@ async def send_signal_to_process(
...
             if signal not in ["SIGSTOP", "SIGCONT"]:
                 _, raw_pid, _ = await ops_test.juju(
-                    "ssh", "--container", "postgresql", unit_name, "pgrep", "-x", process
+                    "ssh", "--container", "postgresql", unit_name, "pgrep", opt, process
                 )
 
-                # If something was returned, there is a running process.
-                if len(raw_pid) > 0:
+                # If the same output was returned, process was not restarted.
+                if raw_pid == old_pid:
```
Comment (dragomirp) [inline tests/integration/ha_tests/helpers.py:708]: Pebble might have restarted the process by the time we ask, so we might receive different pids.

File: metadata.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ resources:
   postgresql-image:
     type: oci-image
     description: OCI image for PostgreSQL
-    upstream-source: ghcr.io/canonical/charmed-postgresql:14.7-22.04_edge
+    upstream-source: ghcr.io/canonical/charmed-postgresql@sha256:6e620a65427b7bb98df6eb85170cfead68d6d9443aee7d136a55c1d737cfea10
```
Comment (dragomirp) [inline metadata.yaml:29]: Pinning the image digest

File: tests/integration/ha_tests/helpers.py (hunk 1 above)
Comment (marceloneppel) [reply tests/integration/ha_tests/helpers.py:708]: Great idea.

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Pull request merged**